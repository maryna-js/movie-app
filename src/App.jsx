import React from 'react';
import { RoutePaths } from './components/routes/Routes'
import './App.css';

const App = () => {
  return (
    <RoutePaths/>
  );
};

export default App;
