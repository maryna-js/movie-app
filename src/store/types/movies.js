export const LOADING = '/fetch/movies/LOADING';
export const LOADED = '/fetch/movies/LOADED';
export const ERROR = '/fetch/movies/ERROR';
