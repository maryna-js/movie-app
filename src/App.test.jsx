import React from 'react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import movies from './__mocks__/mockMovies.json';
import App from './App';

const mockStore = configureStore([]);

describe('App', () => {
    const store = mockStore({
        movies: { movies },
    });
    store.dispatch = jest.fn();
    const history = createMemoryHistory();

    it('Should render main page', () => {
        history.push('/');
        const getComponent = () =>
            render(
                <Router navigator={history}>
                    <Provider store={store}>
                        <App />
                    </Provider>
                </Router>
            );
        expect(getComponent({}).asFragment()).toMatchSnapshot();
    });
});

