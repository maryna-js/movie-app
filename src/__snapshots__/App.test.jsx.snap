// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render main page 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-white flex justify-center"
  >
    <div
      class="border shadow-xl rounded-lg w-11/12 h-full p-4 m-4"
    >
      <div
        class="flex flex-col items-start border rounded w-full h-full my-4 h-auto md:h-36"
      >
        <div
          class="flex flex-col md:flex-row justify-between items-center w-full relative"
        >
          <div
            class="flex w-full md:w-1/2"
          >
            <img
              alt="Star Wars"
              class="w-20 md:w-24 h-32 md:h-36 object-cover"
            />
            <div
              class="flex flex-col px-5 md:pl-7 pt-4 md:pt-6"
            >
              <a
                href="/movie/tt0076759"
              >
                <h2
                  class="font-bold text-base md:text-lg text-blue-800 h-12 md:h-14 overflow-hidden"
                >
                  Star Wars
                </h2>
              </a>
              <div
                class="text-blue-700 font-light text-sm pt-2"
              >
                movie
              </div>
              <div
                class="text-blue-600 font-light text-xs pt-2"
              >
                1977
              </div>
            </div>
          </div>
          <div
            class="w-full md:w-64 flex items-center justify-start md:justify-center px-1 py-2 md:p-4"
          >
            <div
              class="mx-2"
            >
              <div
                class="w-4 md:w-8"
              />
            </div>
            <div
              class="text-sm md:text-lg font-medium text-blue-700 pl-2 md:pl-4"
            >
              5.8 k
            </div>
            <div
              class="ml-4 md:ml-8"
            >
              <div
                class="w-4 md:w-8"
              />
            </div>
            <div
              class="text-sm md:text-lg font-medium text-blue-700 pl-2 md:pl-4 "
            >
              4.30
            </div>
          </div>
          <button
            class="text-blue-900 font-bold text-lg mr-6 absolute md:relative right-4 md:right-0 top-20 md:top-0"
            data-testid="collapse"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
